app/services/course/conditional/conditional_resolving_service.rb

```diff
@@ -0,0 +1,48 @@
+class Course::Conditional::ConditionalResolvingService
+  class << self
+    # Resolve the conditionals for the given course user
+    #
+    # @param[CourseUser] course_user The course user with conditionals to be resolved
+    delegate :resolve, to: :new
+  end
+
+  # Resolve the conditionals for the given course user
+  #
+  # @param[CourseUser] course_user The course user with conditionals to be resolved
+  def resolve(course_user)
+    # Retrieve the conditional dependency graph for the course user
+    conditional_graph = conditional_graph_for(course_user.course)
```

Comment:
Some of these variables, like course_user and course can be instance variables, which are assigned in an initialize. The Service object only is accessible from one thread, and it is encapsulated, so that is fine. Saves you passing objects around as arguments